src/number_theory/legendre_symbol/quadratic_reciprocity.lean

```diff
@@ -39,6 +39,23 @@ namespace zmod
 
 variables (p q : ℕ) [fact p.prime] [fact q.prime]
 
+/-- A prime `p` that is not 2 satisfies `p % 2 = 1` -/
+lemma nat.prime.mod_two_eq_one_of_not_eq_two {p : ℕ} [pp : fact p.prime] (hp : p ≠ 2) :
+  p % 2 = 1 :=
+or.dcases_on (nat.prime.eq_two_or_odd pp.1)
+             (λ (h : p = 2), absurd h hp) (λ (h : p % 2 = 1), rfl.mpr h)
+
+/-- A prime `p` satisfies `p % 2 = 1` if and only if `p ≠ 2`. -/
+lemma nat.prime.mod_two_eq_one_iff_not_eq_two {p : ℕ} [pp : fact p.prime] : p % 2 = 1 ↔ p ≠ 2 :=
+begin
+  split,
+  { intros h hf,
+    simp [hf] at h,
+    exact h, },
+  { intro h,
+    exact nat.prime.mod_two_eq_one_of_not_eq_two h, },
+end
```

Comment:
I think we should only add the `iff` version.
```suggestion
/-- A prime `p` satisfies `p % 2 = 1` if and only if `p ≠ 2`. -/
lemma nat.prime.mod_two_eq_one_iff_ne_two : p % 2 = 1 ↔ p ≠ 2 :=
begin
  refine ⟨λ h hf, _, (nat.prime.eq_two_or_odd $ fact.out p.prime).resolve_left⟩,
  rw hf at h,
  simpa using h,
end
```
(Sometimes we have all three lemmas of the form `foo.of_bar`, `bar.of_foo`, `foo_iff_bar` but usually this is just to enable dot notation, which does not apply here.)

I also think this lemma should probably be placed next to `nat.prime.eq_two_or_odd` in `data/nat/prime.lean`.

Reply:
OK. I removed the "_of_" version, changed the name from "_not_eq_two" to "_ne_two" and moved the lemma to `data.nat.prime` (and then it took ages to build, since `data.nat.prime` seems to be imported by most of mathlib...).